src/main/kotlin/org/jitsi/nlj/dtls/DtlsUtils.kt

```diff
@@ -251,6 +254,52 @@ class DtlsUtils {
             }
             return buf.toString()
         }
+
+        /*
+         * Copied from TlsContext#exportKeyingMaterial and modified to work with
+         * an externally provided masterSecret value.
+         */
+        fun exportKeyingMaterial(
+            context: TlsContext,
+            asciiLabel: String,
+            context_value: ByteArray?,
+            length: Int,
+            masterSecret: TlsSecret
+        ): ByteArray {
+            if (context_value != null && !TlsUtils.isValidUint16(context_value.size)) {
+                throw IllegalArgumentException("'context_value' must have a length less than 2^16 (or be null)")
+            }
+            val sp = context.securityParameters
+            val cr = sp.clientRandom
+            val sr = sp.serverRandom
+
+            var seedLength = cr.size + sr.size
+            if (context_value != null) {
+                seedLength += (2 * context_value.size)
```

Comment:
Should this read `2 + context_value.size`?

Reply:
yes! nice catch